### ISSUE TITLE
Separate MenuItem from PhetMenu into PhetMenuItem.js

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#393]
Currently it looks like: 
```js

    var menuItem = new Node( {
      cursor: 'pointer',
      textDescription: text + ' Button'
    } );
    menuItem.addChild( highlight );
    menuItem.addChild( textNode );

    textNode.left = highlight.left + LEFT_X_MARGIN + CHECK_OFFSET; // text is left aligned
    textNode.centerY = highlight.centerY;

    menuItem.startedCallbacksForFiredEmitter = new Emitter();
    menuItem.endedCallbacksForFiredEmitter = new Emitter();

    menuItem.addInputListener( {
      enter: function() { highlight.fill = HIGHLIGHT_COLOR; },
      exit: function() { highlight.fill = null; }
    } );
```

and many other members added on. This is messy.

**Comment by zepumph** [comment]
@pixelzoom could you look over this one at your convenience.

**Comment by pixelzoom** [comment]
I fixed some basic formatting in the above commit, let me know if you have any questions about that. @zepumph you seem to have a tendency to leave multiple blank lines hanging around. Are you using the PhET Style in IDEA, and are you formatting your code?

Other things that need to be addressed:

(1) There is no documentation in the header comment.  The header comment should provide a short description of the file and its responsibilities. (See TODO in code.)

(2) Constructor parameters are not documented and have no type expressions. (See TODO in code.)

(3) `Node` constructor is called with `options`, and then `mutate` is called later with additional options.  Options can be passed to the supertype by _either_ method, but (in general) should not be passed by both methods in a constructor. 

(4) The reason for (3) is so you don't do what you've done here with the `tandem` option. To `Node.call`, you're passing `options.tandem.createSupertypeTandem()`.  To `mutate`, you're passing `options.tandem`.  Those are 2 different tandems, so this smells like a problem.

(5) The pattern for implementing `dispose` does not match PhET convention.  See `sun.Carousel` for an example of the correct pattern. `dispose` should be a prototype function that calls an instance function (in this case `disposeMenuItem`).

**Comment by pixelzoom** [comment]
I put TODO items in the code for the problems noted in https://github.com/phetsims/joist/issues/393#issuecomment-278717098.

**Comment by zepumph** [comment]
@pixelzoom thank you so kindly for all of your suggestions. I think that I covered them all, and things seem to be working well (with documentation which is always a plus). @samreid would you mind looking at my most recent commit (I take out the call to the supertype tandem and just have the mutate call). I know that we recently changed it away from passing the supertype in Node.call and then passing in the real tandem in mutate, but I would love to be double checked. 

Otherwise I think this issue is ready to be closed. Thank you for all the help!

**Comment by samreid** [comment]
The changes look good and tested OK, I added a few cosmetic improvements in the above commits, please review them and let me know if you have any questions about my changes.

**Comment by zepumph** [comment]
Makes sense, thank you. Closing.

**Comment by zepumph** [comment]
After looking at some comments from #394, I made tandem a required argument, along with everything necessary for a MenuItem. @samreid will you take a look at this commit. It is passing phet/phet-io tests and works for my general using of the menu.

**Comment by pixelzoom** [comment]
As mentioned in https://github.com/phetsims/joist/issues/394#issuecomment-283150742:

> In my quick glance at MenuItem(#393) ... It has no dependencies on other joist code, and looks like it could be generally useful to client. So perhaps it should move to sun, and tandem should be an option.

**Comment by zepumph** [comment]
From https://github.com/phetsims/joist/issues/394#issuecomment-283171458:

>For MenuItem I will move it to sun and tandem will be an option.

**Comment by zepumph** [comment]
@pixelzoom could you look over this one too. Home stretch.

**Comment by pixelzoom** [comment]
@zepumph I made one change: renamed `var CHECK_MARK_NODEHolder` to `var checkMarkWrapper`.  The name `CHECK_MARK_NODEHolder` violates PhET (and general) conventions for a var name, it's a weird combination of constant and camel case convention. I get why you named it that.  And I understand that this Node is needed because `CHECK_MARK_NODE` is a constant, and may therefore appear in more than one place in the scene graph.  We usually refer to this type of Node as a "wrapper node", which is why I renamed it `checkMarkWrapper`.   

If it was me, I would probably ditch `CHECK_MARK_NODE` and its wrapper, and just create a `FontAwesomeNode( 'check_without_box' )` for each `MenuItem` instance.  We're unlikely to have so many MenuItems that this constant will make a difference in memory footprint.  And the added complexity of the wrapper hardly seems worth the savings in this case.  But up to you...

Back to @zepumph for review.
